packages/sanity/src/cli/actions/exec/execScript.ts

```diff
@@ -36,23 +34,17 @@ const execScript: CliCommandAction<ExecFlags> = async function execScript(args,
     throw new Error(`${scriptPath} does not exist`)
   }
 
-  const esbuildPath = path.resolve(__dirname, './esbuild.js')
-  const browserEnvPath = path.resolve(__dirname, './registerBrowserEnv.js')
-  const configClientPath = path.resolve(__dirname, './configClient.js')
-  const baseArgs = mockBrowserEnv ? ['-r', browserEnvPath] : ['-r', esbuildPath]
-  const tokenArgs = withUserToken ? ['-r', configClientPath] : []
-
-  const nodeArgs = [...baseArgs, ...tokenArgs, scriptPath, ...args.extraArguments]
-
-  const proc = spawn(process.argv[0], nodeArgs, {
-    stdio: 'inherit',
-    env: {
-      // eslint-disable-next-line no-process-env
-      ...process.env,
-      SANITY_BASE_PATH: workDir,
-    },
-  })
-  proc.on('close', process.exit)
+  if (mockBrowserEnv) {
+    await import('./registerBrowserEnv')
+  } else {
+    await import('./esbuild')
+  }
+
+  if (withUserToken) {
+    await import('./configClient')
+  }
+
+  await import(scriptPath)
```

Comment:
I think these changes make a lot of sense – using dynamic imports to split up the runtime. But does this mean we are no longer running these processes in child threads?

Reply:
> How do you feel about not running these in child processes any more? We _could_ solve that by adding entry points for each of the child process scripts.

Don't feel great about it - so your suggested entry point approach looks great to me! I'll take a stab at it.